### ISSUE TITLE
Current dataset is removed after refreshing

### ISSUE THREAD

**Issue by arielsvn** [AlexsLemonade/refinebio-frontend#691]
![2019-07-12 11 15 33](https://user-images.githubusercontent.com/1882507/61138971-6a7ae780-a496-11e9-95e8-a7e9c25c2953.gif)

**Comment by wvauclain** [comment]
It looks like this happens because `dataset-id` is somehow removed from Local Storage when you click on the `My Dataset` button.

**Comment by wvauclain** [comment]
Though this also seems to happen when you refresh the page, which causes a related bug. You can refresh once on the search page without losing your dataset, but it gets lost when you refresh for the second time.

**Comment by davidsmejia** [comment]
If you add 
```javascript
console.error('id', action.data.id, 'datasetid', action.data.dataSetId);
``` 
to ```src/state/download/reducer.js```
You can see that the two values flip flop depending on the type of state change that occurs.
Which ultimately makes the persist middleware remove the key which makes you lose your dataset session as @wvauclain pointed out.

**Comment by wvauclain** [comment]
Fixed by #694
